policies/registry.rego

```diff
@@ -0,0 +1,42 @@
+package before_resolution
+
+# This file enforces policies requiring all attributes to be defined within
+# a semantic convention "registry".  This is a naming/structure convention
+# used by semantic conventions.
+
+# Helper to create attribute registry violations.
+attr_registry_violation(violation_id, group_id, attr_id) = violation {
+	violation := {
+		"id": violation_id,
+		"type": "semconv_attribute",
+		"category": "attribute_registry",
+		"group": group_id,
+		"attr": attr_id,
+	}
+}
+
+# We only allow attribute groups in the attribute registry.
+deny[attr_registry_violation("registry_must_be_attribute_group", group.id, "")] {
```

Comment:
I'm playing with collisions and would like to put something like `Attribute 'messaging.client_id' has the same constant name 'messaging_client_id' as ["messaging.client.id"]`  somewhere.

There seem to be no other place to put it except `violation.id`, so I wonder if we could use violation id as an error message.

Reply:
Do we really need jinja for it? 😢

Reply:
See https://github.com/open-telemetry/semantic-conventions/pull/1209 for the example

Reply:
Jinja just lets us control the error message.  If you're happy with the default, you can use it.

I wanted to do better than the default, which I don't think is super readable, so I made a format.  What I like is we don't need weaver changes to update error messages or policy violations.

If you'd rather use sprintf in rego, that's fine.  I still think the error I show above is more readable.

ALSO - once MINIJINJA supports `groupby` from JINJA we can actually sort error messages by file instance (for before_resolution), etc. etc.

Reply:
The concern is that policies will have inconsistent output - some will return `registry_must_be_attribute_group` as violation id and other will return free-form error description. The more complicated policies are, the more likely that `Violation` would not have properties to describe all  details.

E.g. in case of collisions I have to return the problematic attribute name and a list of all collisions.

Since we'll have the same makefile check for all policies and the same jinja template, the template would need to treat the output differently depending on something. It feels like a big complication to anyone writing policies - they also need to adjust output formatting.

So I think we need consistency and given that all we need is for humans to read error messages, having something simple and obvious would make things easier for 
- policy authors
- reviewers
- someone who tries to decode butchered error message

Reply:
We discussed this in the tooling SiG.  I think you have the right requirements here, specifically:

1. We want the ability to define both POLICY and HUMAN-READABLE ERROR in the same location
2. I'd like to see some weaver improvements in this area to give us the flexibility we need.

So going forward, we're going to investigate better "string creation" in Rego *and* a better Violation structure in weaver that can take advanatage of that.

I've moved to having the readability of error messages rely on just the violation ID for now.  Once weaver supports better errors, we can improve this.